Retention time correction and additional peak fitting algorithm

adwolfer opened the pull request:
Add retention time correction (RANSAC algorithm) and Exponentially Modified Gaussian (EMG) peak fitting

adwolfer committed:
bump to 1.3.0 devel version

adwolfer committed:
Bump to 1.3.0 devel version

Gscorreia89 committed:
Ransac algorithm (#20)

* rtCorrection_utils.R, add back placeholder for retention time correction

* rt_correction interface work

* update to R4.0, minor style corrections

* # inst: scripts/make_unittest_data.R added
Sync branch with develop

* rtCorrection_utils:
- Rt correction polynomial model added
- RANSAC algorithm added
methods_peakPantheRAnnotation.R:
- Rt correction method modified

* methods_peakPantheRAnnotation.R:
- Rt correction method modified
- diagnostic plot added

* methods_peakPantheRAnnotation.R:
- small changes
- applyRTCorrection unitests

* methods_peakPantheRAnnotation.R:
- Change to figure axes label
tests:
- test_peakPantheRAnnotation_retentionTimeCorrection-method.R added - tests for peakPantheRAnnotation object method and plot

* methods_peakPantheRAnnotation.R:
- Documentation fix

* parallel-annotation.Rmd:
- add retention time correction example

* BioC style corrections (T/F, stats::)

* fix travis allowed failures

* updated vignettes

* documentation correction `retentionTimeCorrection` method

* sort line > 80 character in code

* sort line > 80 character in man and vignette

* fix indent to multiples of 4

* fix function length:
add ggplot variables from rtCorrection diag plot to globalVars

* fix function length (in progress):
modifications in method and related unittests and correction utils

* fix function length (in progress):
modifications to RANSAC

* documentation change
* RtCorrection_method - fix defaults and parameters in method definition

Co-authored-by: Arnaud M. Wolfer <adwolfer@gmail.com>

jaketmp commented:
Looks like you have a linker issue on the OSX build, adding:
`if [ ${TRAVIS_OS_NAME} = 'osx' ]; then rm '/usr/local/bin/gfortran'; fi` 
before
`if [ ${TRAVIS_OS_NAME} = 'osx' ]; then brew install libgit2; fi`
in the travis config should help.

adwolfer committed:
updated NEWS

Gscorreia89 committed:
Feature/emg gaussian (#22)

* curveFit_utils.Rmd:
- added cerf function and equations for emgGaussian
* findTargetFeatures, peakPantheR_parallelAnnotation and peakPantheR_singleFileSearch.R:
- curveModel arguments to support emgGaussian
* add new .rD man files
add osx fortran clause to travis
* move to travis.com in README
* - Modifications to parallel and singleFilesearch for new model
Added unittests for emgGaussian lineshape
* - fitCurve test_added
* New travis setup: release allowed to fail and biocCheck does not check version number in release
* add predictCurve test with emgGaussian
* travis modification to terminate OSX builds
* travis no codecov on OSX
* added mention of curveModel to vignette docs and function calls

Co-authored-by: Arnaud M. Wolfer <adwolfer@gmail.com>

adwolfer committed:
increased codecov coverage

adwolfer commented on man/findTargetFeatures.Rd:
```diff
@@ -18,7 +25,7 @@ with signal as row and retention time ('rt'), mass ('mz') and intensity
 
 \item{curveModel}{(str) Name of the curve model to fit (currently
-\code{skewedGaussian})}
+\code{skewedGaussian} and \code{emGaussian})}
```
emgGaussian?

adwolfer committed:
curveModel test coverage

adwolfer commented:
ongoing [issue](https://github.com/Bioconductor/BiocStyle/issues/78) between `knitr` and `BiocStyle` prevents the generation of vignettes, status can be monitored [here](https://github.com/Bioconductor/BiocStyle/pull/79)

adwolfer committed:
travis BiocStyle

adwolfer committed:
add Suggest to travis

adwolfer committed:
increased codecov coverage
